index.d.ts

```diff
@@ -93,8 +93,8 @@ export function combineReducers<S>(reducers: ReducersMapObject<S>): Reducer<S>;
  * transform, delay, ignore, or otherwise interpret actions or async actions
  * before passing them to the next middleware.
  */
-export interface Dispatch<S> {
-    <A extends Action>(action: A): A;
+export interface Dispatch<A extends Action = Action> {
```

Comment:
I'm not sure this is correct. Redux store is able to dispatch any possible action, so we can't constrain it to a specific subtype.

Also, here's the rationale for making state type a type parameter for `Dispatch`: https://github.com/reactjs/redux/pull/1537
However, for the whole year, I've only seen it being used in `redux-thunk`. Moreover, it would give type-checking benefits only when `dispatch` is called directly via `store.dispatch(...)`, but if we e.g. use `connect` from `react-redux`, we'd still have to declare state type manually.

In light of the above, I tend to agree that we should consider removing the `S` type parameter from `Dispatch`.